### PR TITLE
feat💥: Improve invites (see desc.)

### PR DESCRIPTION
**Pull request by leestarb**
## What type of pull request is this?

- [ ] Non-breaking code change
- [x] Breaking code change
- [ ] Documentation change/addition

## Description

Awesome improvements for the `Invite` object

## Changes

- Add delete ability
- Fix invite's users properties 💥
- Implement `__str__` and `__repr__`

## Checklist

- [x] I've formatted my code with [Black](https://black.readthedocs.io/en/stable/)
- [x] I've ensured my code works on `Python 3.10.x`
- [x] I've tested my code

### CODE REVIEW

File: dis_snek/api/http/http_requests/invites.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+from typing import Any
+
+from dis_snek.client.const import MISSING, Absent
+from ..route import Route
+
+
+class InviteRequests:
```
Comment (Wolfhound905) [inline dis_snek/api/http/http_requests/invites.py]: This file is never used, and you never added it to 
https://github.com/Discord-Snake-Pit/Dis-Snek/blob/e85c04abf42650958a2a750674650cccbc6456f1/dis_snek/api/http/http_client.py#L52

I also don't think this is needed

File: dis_snek/api/http/http_requests/invites.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    request: Any
+
+    async def delete_invite(self, invite_code: str, reason: Absent[str] = MISSING) -> dict:
```
Comment (Wolfhound905) [inline dis_snek/api/http/http_requests/invites.py]: This is a duplicate of a function we already have in https://github.com/Discord-Snake-Pit/Dis-Snek/blob/e85c04abf42650958a2a750674650cccbc6456f1/dis_snek/api/http/http_requests/channels.py#L241
I do not think this file is needed.

**Review by Wolfhound905 — CHANGES_REQUESTED** [review]

File: dis_snek/models/discord/invite.py [hunk 3]
```diff
@@ -16,47 +16,51 @@
...
 @define()
 class Invite(ClientObject):
-    code: str = field()
+    code: str = field(repr=True)
 
     # metadata
-    uses: int = field(default=0)
+    uses: int = field(default=0, repr=True)
     max_uses: int = field(default=0)
     created_at: Timestamp = field(default=MISSING, converter=optional_c(optional_timestamp_converter))
```
Comment (benwoo1110) [inline dis_snek/models/discord/invite.py]: I think this can be included in repr too?

**Review by benwoo1110 — APPROVED** [review]

**Comment by leestarb** [comment]
Please look at https://discord.com/channels/870046872864165888/870046872864165893/934362064737140796

**Review by LordOfPolls — APPROVED** [review]

**Pull request merged**